tests/webTables.spec.ts

```diff
@@ -0,0 +1,182 @@
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
+        }
+    }
+
+    await expect(specialtyInput).toHaveValue('dermatology')
+    await specialtyInput.clear()
+    await specialtyInput.fill('surgery')
+    await page.getByText('Update').click()
+})
+
+test('validate specialty lists', async ({ page }) => {
+    //1. Select the SPECIALTIES menu item in the navigation bar
+    await page.getByText('Specialties').click()
+    await page.waitForSelector('tbody')
+    //2. On the Specialties page, select "Add" button. Type the new specialty "oncology" and click "Save" button
+    await page.getByRole('button', { name: 'Add' }).click()
+    await page.locator('#name').fill('oncology')
+    await page.getByText('Save').click()
+    await expect(page.locator('tbody tr').last().getByRole('textbox')).toHaveValue('oncology')
+    //3. Extract all values of specialties and put them into the array.
+    const allSpecialtyInputs = await page.locator('tbody tr').getByRole('textbox').all()
+    const allSpecialties: string[] = []
+
+    for (const input of allSpecialtyInputs) {
+        allSpecialties.push(await input.inputValue())
+    }
+
+    //4. Select the VETERINARIANS menu item in the navigation bar, then select "All"
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
+    //5. On the Veterinarians page, locate the "Sharon Jenkins" in the list and click "Edit" button
+    await page.getByRole('row', { name: 'Sharon Jenkins' }).getByText('Edit Vet').click()
+    //6. Click on the Specialties drop-down menu. Extract all values from the drop-down menu to an array
+    await page.locator('.dropdown-display').click()
+    //7. Add the assertion that array of specialties collected in the step 3 is equal the the array from drop-down menu
+    const allValuesFromDropDownMenu = await page.locator(".dropdown-content label").allTextContents()
+    expect(allSpecialties).toEqual(allValuesFromDropDownMenu)
+    //8. Select the "oncology" specialty and click "Save vet" button
+    await page.getByRole('checkbox', { name: 'oncology' }).check()
+    await page.locator('.dropdown-display').click()
+    await page.getByText('Save Vet').click()
+    //9. On the Veterinarians page, add assertion, that "Sharon Jenkins" has specialty "oncology"
+    await expect(page.getByRole('row', { name: 'Sharon Jenkins' }).getByRole('cell').nth(1)).toHaveText('oncology')
+    //10. Navigate to SPECIALTIES page. Click "Delete" for "oncology" specialty
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await page.locator('tbody tr').last().getByRole('button', { name: "Delete" }).click()
+    //11. Navigate to VETERINARIANS page. Add assertion that "Sharon Jenkins" has no specialty assigned
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
```

Comment:
Earlier in your code in line 122, you have used a simpler version of the locator than this version. Why?